lib/process-services-cloud/src/lib/form/components/widgets/dropdown/dropdown-cloud.widget.ts

```diff
@@ -128,6 +131,22 @@ export class DropdownCloudWidgetComponent extends WidgetComponent implements OnI
     }
 
     ngOnInit() {
+        /*
+            We can have a lot of 'control.setValue' caused by form rules events
+            e.g. every time if we focusin/focusout etc. we are calling a setValue.
+        */
+        this.debounceSetValue.pipe(debounceTime(DROPDOWN_CLOUD_WIDGET_SET_VALUE_DEBOUNCE), takeUntilDestroyed(this.destroyRef)).subscribe(() => {
+            if (Array.isArray(this.field.value)) {
+                this.dropdownControl.setValue(this.field?.value, { emitEvent: false });
+            } else if (this.field?.value && typeof this.field?.value === 'object') {
+                this.dropdownControl.setValue({ id: this.field?.value.id, name: this.field?.value.name }, { emitEvent: false });
+            } else if (this.field.value === null) {
+                this.dropdownControl.setValue(this.field?.value, { emitEvent: false });
+            } else {
+                this.dropdownControl.setValue({ id: this.field?.value, name: '' }, { emitEvent: false });
+            }
```

Comment:
maybee
```suggestion
            let newValue; 
            if (Array.isArray(this.field.value)) {
                newValue = this.field?.value;
            } else if (this.field?.value && typeof this.field?.value === 'object') {
                newValue = { id: this.field?.value.id, name: this.field?.value.name };
            } else if (this.field.value === null) {
                newValue = this.field?.value;
            } else {
                newValue = { id: this.field?.value, name: '' };
            }
            this.dropdownControl.setValue(newValue, { emitEvent: false });
```
Could even move the new value logic to a separate private function and use switch/case with returns

Reply:
updated